var/spack/repos/builtin/packages/py-pymatgen/package.py

```diff
@@ -15,21 +15,43 @@ class PyPymatgen(PythonPackage):
     homepage = "http://www.pymatgen.org/"
     pypi = "pymatgen/pymatgen-4.7.2.tar.gz"
 
+    version('2022.0.14', sha256='03d24ebafc21becab376c26de09437dd5cfb7bda7099046194e95b4c9fa35209')
+    version('2021.3.5',  sha256='de1f65dd3f4b1eb1626062865da29aa556c1796d93b9289135e0dc64a0adb286')
     version('4.7.2', sha256='e439b78cc3833a03963c3c3efe349d8a0e52a1550c8a05c56a89aa1b86657436')
     version('4.6.2', sha256='f34349090c6f604f7d402cb09cd486830b38523639d7160d7fd282d504036a0e')
 
-    extends('python', ignore='bin/tabulate')
+    # starting from version 2021.1.1, pymatgen requires Python >=3.7
+    extends('python@:3.6',               ignore='bin/tabulate',  when='@:4.7.2')
```

Comment:
```suggestion
    extends('python',               ignore='bin/tabulate',  when='@:4.7.2')
```
There are no restrictions on the Python version for older releases